gpus/python-gpu-example.html.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Python GPU Dev Machine 
+layout: docs
+nav: firecracker
+---
+
+This is a worked example for a CUDA-enabled Python template environment on a Fly GPU Machine, for working with ML models.
+
+We'll start with a minimal Ubuntu Linux, add a non-root user, and set up a Python virtual environment for a project, with Jupyter Notebook installed. NVIDIA libraries needed for the project can be installed to the persistent disk as needed, using pip.
+
+## Deployment to Fly.io
+### Create a Fly App
+```cmd
+fly apps create <app-name>
+```
+
+### Clone the example repo
+```cmd
+git clone git@github.com:fly-apps/python_gpu_example.git && cd python_gpu_example
+```
+
+### Edit app configuration in `fly.toml`
+* At the least, change `app` to match the name of the app you just created.
+* Edit `primary_region` if wanted -- make sure to choose a region in which GPU Machines are available. Check the [docs](https://fly.io/docs/gpus/gpu-quickstart/) for GPU regions.
+* Optionally, change the NONROOT_USER `build.arg` -- if you do, also edit the `destination` of the volume mount in `mounts` to match.
+* Optionally, tweak `swap_size_mb`.
+
+### Deploy
+
+```cmd
+ fly deploy
+```
+ 
+ The `fly deploy` command launches the app, provisioning initial resources on first run. In this case, it creates one Fly Machine VM and one Fly Volume. When an app is configured to use a volume, `fly deploy` does not create a redundant or standby Machine.
+
+## Using the Machine
+
+### Jupyter notebook
+
+The easiest way to visit a private Fly App with the browser is with `fly proxy` command, which proxies a local port to a Machine.
+
+Run `fly logs` to find a line like 
+
+```
+http://127.0.0.1:8888/tree?token=c5fe8a87d8c00dd16637f0d4a1d8df7e3590c6a6064bbb6b
+```
+
+from Jupyter. Visit that link in the browser and you can start up a notebook.
+
+
+### Terminal
+
+Connect to the Machine using `fly ssh console`. To activate the configured venv, do the following:
+
+```
+# su pythonuser
+$ cd ~/project/
+$ source ~/project-venv/bin/activate 
+```
+
+Then you can install new pip packages to the persistent volume, download models, and run the Python REPL or execute scripts.
+
+To deactivate the venv, type `deactivate`. To drop back to the root user, hit <kbd>CTRL-D</kbd>.
+
+As the root user, you can use apt to manage system-wide software. Anything you install with apt is installed to the Machine's root file system, which means it disappears when the Machine next shuts down, so if you find yourself doing this, remember to add them to the Dockerfile, ready to be built into the image on the next deployment.
```

Comment:
```suggestion
As the root user, you can use apt to manage system-wide software. Anything you install with apt is installed to the Machine's root file system, which means it disappears when the Machine next shuts down, so remember to add them to the Dockerfile, ready to be built into the image on the next deployment.
```

Reply:
I like it like this. It's a nudge to notice if you're doing it. (Sneak it out if you ever have to edit the doc I guess)